Tests/MongoSwiftTests/APMTests.swift

```diff
@@ -0,0 +1,240 @@
+        }
+        try await client.db("admin").runCommand(["ping": 1])
+        try await client.close()
+
+        // Ensure the task returned with proper number of events
+        let numEvents = try await eventTask.value
+        expect(numEvents).to(equal(4))
+    }
+
+    func testCommandStreamBufferingPolicy() async throws {
+        let elg = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer { elg.syncShutdownOrFail() }
+        let client = try MongoClient.makeTestClient(eventLoopGroup: elg)
+        let insertsDone = NIOAtomic<Bool>.makeAtomic(value: false)
+
+        let taskBuffer = Task { () -> Int in
+            let stream = client.commandEventStream()
+            try await assertIsEventuallyTrue(description: "inserts done") {
+                insertsDone.load()
+            }
+            var i = 0
+
+            for try await _ in stream {
+                i += 1
+            }
+            return i
+        }
+        for _ in 1...60 { // 120 events
+            try await client.db("trialDB").collection("trialColl").insertOne(["hello": "world"])
+        }
+        insertsDone.store(true)
+        try await client.close()
+
+        let eventTypes1 = try await taskBuffer.value
+
+        // Cant check for explictly 100 because of potential load balancer events getting added in during for loop
+        expect(eventTypes1).to(beLessThan(120))
+    }
+
+    // Actor to handle array access/modification in an async way
+    actor EventArray {
+        var eventTypes: [[EventType]] = []
+        var tasks: [Task<Bool, Error>] = []
+
+        func appendType(eventType: EventType, index: Int) {
+            self.eventTypes[index].append(eventType)
+        }
+
+        func appendTasks(task: Task<Bool, Error>) {
+            self.tasks.append(task)
+        }
+
+        func newTask() {
+            self.eventTypes.append([])
+        }
+
+        func getResults() async throws -> [Bool] {
+            var output: [Bool] = []
+            for elt in self.tasks {
+                try await output.append(elt.value)
+            }
+            return output
+        }
+    }
+
+    /// Helper that tests kicking off multiple streams concurrently
+    fileprivate func concurrencyHelper<T>(f: @escaping (MongoClient) -> T) async throws
+        where T: AsyncSequence, T.Element: StreamableEvent
+    {
+        let eventArray = EventArray()
+        let elg = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer { elg.syncShutdownOrFail() }
+        let client = try MongoClient.makeTestClient(eventLoopGroup: elg)
+
+        // Kick off 5 streams
+        for i in 0...4 {
+            let task = Task { () -> Bool in
+                await eventArray.newTask()
+                for try await event in f(client) {
+                    await eventArray.appendType(eventType: event.type, index: i)
+                }
+
+                return true
+            }
+            await eventArray.appendTasks(task: task)
```

Comment:
Sorry, Im a little confused here - did you mean make the `appendTasks` and `newTask` methods happen together? This is a little challenging because the `newTask` method has to happen inside the task to ensure all the arrays are init'd properly (tried combining with `appendTasks` but resulted in timeout bc the arrays werent inited in time), but the `appendTasks` have to happen with the task already defined.